sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -991,6 +999,16 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
             Row(null, null, null, "]") :: Nil
         )
 
+        // Check if corrupt records are dropped.
```

Comment:
It shows the results below:
```
        sql(
          """
            |SELECT a, b, c, _unparsed
            |FROM jsonTableWithDropMalformed
          """.stripMargin).show()
```
```
+---------+-------+-------+-------+
|_unparsed|      a|      b|      c|
+---------+-------+-------+-------+
|     null|str_a_4|str_b_4|str_c_4|
+---------+-------+-------+-------+
```

However, I think as you said I might have to make them separate. Yea. It is a bit confusing.